cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -243,6 +248,19 @@ func (s *CMServer) Run(_ []string) error {
 	}
 	pvRecycler.Run()
 
+	var rootCAProvider func() []byte
+
+	if s.RootCAFile != "" {
+		rootCA, err := ioutil.ReadFile(s.RootCAFile)
+		if err != nil {
+			return fmt.Errorf("error reading root-ca-file: %v", err)
+		}
+		if _, err := x509.ParseCertificate(rootCA); err != nil {
+			return fmt.Errorf("error parsing root-ca-file: %v", err)
+		}
+		rootCAProvider = func() []byte { return rootCA }
+	}
```

Comment:
if they didn't provide a RootCAFile, should we use the CA data in the `kubeconfig` if possible?